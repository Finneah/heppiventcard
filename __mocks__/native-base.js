jest.mock('native-base', () => {
  return {
    StyleProvider: jest.fn().mockImplementation((x) => x),
  };
});
