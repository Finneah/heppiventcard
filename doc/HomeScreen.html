<!DOCTYPE html>
<html lang="en">

<head>
  
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Heppiventcard HomeScreen</title>

  <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>
  <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
  <script src="./build/entry.js"></script>
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
  <![endif]-->
  <link href="https://fonts.googleapis.com/css?family=Roboto:100,400,700|Inconsolata,700" rel="stylesheet">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <link type="text/css" rel="stylesheet" href="https://jmblog.github.io/color-themes-for-google-code-prettify/themes/tomorrow-night.min.css">
  <link type="text/css" rel="stylesheet" href="styles/app.min.css">
  <link type="text/css" rel="stylesheet" href="styles/iframe.css">
  <link type="text/css" rel="stylesheet" href="">
  <script async defer src="https://buttons.github.io/buttons.js"></script>

  
</head>



<body class="layout small-header">
    <div id="stickyNavbarOverlay"></div>
    

<div class="top-nav">
    <div class="inner">
        <a id="hamburger" role="button" class="navbar-burger" aria-label="menu" aria-expanded="false">
            <span aria-hidden="true"></span>
            <span aria-hidden="true"></span>
            <span aria-hidden="true"></span>
        </a>
        <div class="logo">
            
             
                <a href="index.html">
                    <h1 class="navbar-item">Heppiventcard</h1>
                </a>
            
        </div>
        <div class="menu">
            
            <div class="navigation">
                <a
                    href="index.html"
                    class="link"
                >
                    API Documentation
                </a>
                
                 
                    
                        <a
                            class="link user-link "
                            href="https://github.com/Finneah/heppiventcard"
                        >
                            Github
                        </a>
                    
                        <a
                            class="link user-link "
                            href="https://www.figma.com/file/Ze1jZrjApvyP6sv2e2Dpbb/heppiventcard?node-id=0%3A2932"
                        >
                            Example Application
                        </a>
                    
                
                
            </div>
        </div>
    </div>
</div>
    <div id="main">
        <div
            class="sidebar "
            id="sidebarNav"
        >
            
                <div class="search-wrapper">
                    <input id="search" type="text" placeholder="Search docs..." class="input">
                </div>
            
            <nav>
                
                    <h2><a href="index.html">Documentation</a></h2><div class="category"></div><div class="category"><h2>Component</h2><h3>Namespaces</h3><ul><li><a href="HomeScreenHeader.html">HomeScreenHeader</a></li><li><a href="MemberCard.html">MemberCard</a></li><li><a href="StampCard.html">StampCard</a></li></ul></div><div class="category"><h2>Database</h2><h3>Classes</h3><ul><li><a href="StampCardsModel.html">StampCardsModel</a></li><li><a href="StampCardsSchema.html">StampCardsSchema</a></li><li><a href="StampsModel.html">StampsModel</a></li><li><a href="StampsSchema.html">StampsSchema</a></li><li><a href="UserModel.html">UserModel</a></li><li><a href="UserSchema.html">UserSchema</a></li></ul></div><div class="category"><h2>View</h2><h3>Namespaces</h3><ul><li><a href="HomeScreen.html">HomeScreen</a></li></ul></div>
                
            </nav>
        </div>
        <div class="core" id="main-content-wrapper">
            <div class="content">
                <header class="page-title">
                    <p>Namespace</p>
                    <h1>HomeScreen</h1>
                </header>
                




<section>

<header>
    
        <h2>HomeScreen</h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        
            <div class="description">Startpage of the app.</div>
        

        


<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="screens_HomeScreen.js.html" class="button">View Source</a>
            <span>
                <a href="screens_HomeScreen.js.html">screens/HomeScreen.js</a>, <a href="screens_HomeScreen.js.html#line19">line 19</a>
            </span>
        </p>
    
</dl>


        
    
    </div>
    
    

    

    

    

    

    

    

    
        <div class='vertical-section'>
            <h1>Methods</h1>
            <div class="members">
            
                <div class="member">


    
    <h4 class="name" id="._getStampCards">
        <a class="href-link" href="#._getStampCards">#</a>
        
            
                <span class='tag'>async</span>
            
                <span class='tag'>static</span>
            
        
        <span class="code-name">
            
                _getStampCards<span class="signature">(sectionStampCards)</span><span class="type-signature"></span>
            
        </span>
    </h4>
    

    
    









    <h5>Parameters:</h5>
    
<div class="table-container">
    <table class="params table">
        <thead>
        <tr>
            
            <th>Name</th>
            

            <th>Type</th>

            

            

            <th class="last">Description</th>
        </tr>
        </thead>

        <tbody>
        

            
<tr class="deep-level-0">
  
      <td class="name"><code>sectionStampCards</code></td>
  

  <td class="type">
  
      
<span class="param-type">Object</span>


  
  </td>

  

  

  <td class="description last"></td>
</tr>


        
        </tbody>
    </table>
</div>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="screens_HomeScreen.js.html" class="button">View Source</a>
            <span>
                <a href="screens_HomeScreen.js.html">screens/HomeScreen.js</a>, <a href="screens_HomeScreen.js.html#line86">line 86</a>
            </span>
        </p>
    
</dl>





















</div>
            
                <div class="member">


    
    <h4 class="name" id=".createFirstStampCard">
        <a class="href-link" href="#.createFirstStampCard">#</a>
        
            
                <span class='tag'>async</span>
            
                <span class='tag'>static</span>
            
        
        <span class="code-name">
            
                createFirstStampCard<span class="signature">()</span><span class="type-signature"></span>
            
        </span>
    </h4>
    

    
    













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="screens_HomeScreen.js.html" class="button">View Source</a>
            <span>
                <a href="screens_HomeScreen.js.html">screens/HomeScreen.js</a>, <a href="screens_HomeScreen.js.html#line114">line 114</a>
            </span>
        </p>
    
</dl>





















</div>
            
                <div class="member">


    
    <h4 class="name" id=".deleteAll">
        <a class="href-link" href="#.deleteAll">#</a>
        
            
                <span class='tag'>async</span>
            
                <span class='tag'>static</span>
            
        
        <span class="code-name">
            
                deleteAll<span class="signature">()</span><span class="type-signature"></span>
            
        </span>
    </h4>
    

    
    
    <div class="description">
        delete All for Dev
    </div>
    













<dl class="details">

    

    

    

    

    

    

    

    
        <dt class="important tag-deprecated">Deprecated:</dt><dd><ul class="dummy"><li>NOT IN USE, only for dev</li></ul></dd>
    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="screens_HomeScreen.js.html" class="button">View Source</a>
            <span>
                <a href="screens_HomeScreen.js.html">screens/HomeScreen.js</a>, <a href="screens_HomeScreen.js.html#line67">line 67</a>
            </span>
        </p>
    
</dl>





















</div>
            
            </div>
        </div>
    

    

    
</article>

</section>




            </div>
            
            <footer class="footer">
                <div class="content has-text-centered">
                    <p>Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.6.6</a></p>
                    <p class="sidebar-created-by">
                        <a href="https://github.com/SoftwareBrothers/better-docs" target="_blank">BetterDocs theme</a> provided with <i class="fas fa-heart"></i> by 
                        <a href="http://softwarebrothers.co" target="_blank">SoftwareBrothers - JavaScript Development Agency</a>
                    </p>
                </div>
            </footer>
            
        </div>
        <div id="side-nav" class="side-nav">
        </div>
    </div>
<script src="scripts/app.min.js"></script>
<script>PR.prettyPrint();</script>
<script src="scripts/linenumber.js"> </script>

<script src="scripts/search.js"> </script>

</body>
</html>