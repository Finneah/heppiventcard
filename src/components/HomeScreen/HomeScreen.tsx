import * as React from 'react';

import {
  TourGuideZone, // Main wrapper of highlight component
  useTourGuideController, // hook to start, etc.
} from 'rn-tourguide';

import HomeScreenHeader from '../HomeScreenHeader/HomeScreenHeader';
import MemberCard from '../MemberCard/MemberCard';

import {StampCards, Stamps} from '../../database';

import {StampsModel} from '../../database/Models/StampsModel';

import {strings} from '../../locale/i18n';
import {
  Accordion,
  ScrollView,
  Stack,
  View,
  useColorModeValue,
  Box,
} from 'native-base';
import StampCard from '../StampCard/StampCard';
import {getData, storeData} from '../../storage/AsyncStorage';
import {SectionPart, StampCardType, StampType} from '../../Helper/Types';
import {DEBUGLEVEL} from '../../Helper/Enums';

let stampsModel = new StampsModel();

// type Props = {
//   navigation: any;
// };

/**
 * @category View
 * @namespace HomeScreen
 * @description Startpage of the app.
 */
const HomeScreen: React.FC = (/**{navigation}*/) => {
  let baseCard: SectionPart[] = [];
  let debugLevel: DEBUGLEVEL = DEBUGLEVEL.OFF;

  const [cards, setCards] = React.useState(baseCard);
  const [defaultIndex, setDefaultIndex] = React.useState(0);
  // const {colorMode, toggleColorMode} = useColorMode();
  const mainBgColor = useColorModeValue('white', 'primary.800');
  const accordionTextColorMode = useColorModeValue(
    'primary.600',
    'secondary.400',
  );
  const {
    canStart, // a boolean indicate if you can start tour guide
    start, // a function to start the tourguide
    // stop, // a function  to stopping it
    eventEmitter, // an object for listening some events
  } = useTourGuideController();

  const handleOnStop = (): void => {
    storeData('@tutorialDone', 'true');
  };

  React.useLayoutEffect(() => {
    StampCards.onLoaded(async () => {
      if (debugLevel !== DEBUGLEVEL.OFF) {
        console.info('StampCards loaded', StampCards.data());
      }

      if (StampCards.data().length === 0) {
        await _createFirstStampCard();
      }
    });

    StampCards.onChange(async () => {
      if (debugLevel !== DEBUGLEVEL.OFF) {
        console.info('StampCards changed', StampCards.data());
      }
    });

    Stamps.onLoaded(async () => {
      if (debugLevel !== DEBUGLEVEL.OFF) {
        console.info('Stamps loaded', Stamps.data());
      }

      _getStampCards(StampCards.data());
    });

    Stamps.onChange(async () => {
      if (debugLevel !== DEBUGLEVEL.OFF) {
        console.info('Stamps loaded', Stamps.data());
      }
      _getStampCards(StampCards.data());
    });

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  React.useEffect(() => {
    _startIfPossible();

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [canStart]); // ðŸ‘ˆ don't miss it!

  React.useEffect(() => {
    eventEmitter?.on('stop', handleOnStop);

    return () => {
      eventEmitter?.off('stop', handleOnStop);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  // TUTORIAL
  async function _startIfPossible() {
    var tutorialDone = await getData('@tutorialDone');

    if (canStart && !tutorialDone) {
      // ðŸ‘ˆ Test if you can start otherwise nothing will happen
      if (start) {
        start();
      }
    }
  }
  // TUTORIAL END

  /**
   * @memberof HomeScreen
   * @param {Object} sectionStampCards
   */
  async function _getStampCards(sectionStampCards: StampCardType[]) {
    var sections: SectionPart[] = [];
    var expandedId = getData('@accordionExpandedId');

    if (debugLevel === DEBUGLEVEL.DATA) {
      console.info('_getStampCards', sectionStampCards, Stamps.data());
    }
    for (const stampCard of sectionStampCards) {
      var sectionPart: SectionPart = _setSectionPart(stampCard);
      sections.push(sectionPart);
    }

    setCards(sections.reverse());

    _setSelectedAccordionItem(
      sections,
      typeof expandedId === 'string' ? expandedId : null,
    );
  }

  /**
   * @memberof HomeScreen
   * @description TDB
   * @param stampCard StampCardType
   * @returns SectionPart
   */
  function _setSectionPart(stampCard: StampCardType): SectionPart {
    var content: StampType[] = [];

    if (content.length === 0) {
      content = stampsModel.filterStampsBy({
        stampCard: stampCard.id,
      });
      if (content.length === 0) {
        content = stampsModel.filterStampsBy({
          stampCard_id: stampCard.id,
        });
      }
    }

    var sectionPart: SectionPart = {
      ...stampCard,
      title: stampCard.title,
      content: content,
    };

    return sectionPart;
  }

  /**
   * @memberof HomeScreen
   * @description TDB
   * @param sections
   * @param expandedId
   */
  function _setSelectedAccordionItem(
    sections: SectionPart[],
    expandedId: string | null,
  ) {
    for (let i = 0; i < sections.length; i++) {
      const element = sections[i];
      if (expandedId === element.id) {
        setDefaultIndex(i);
      }
    }
  }
  /**
   * @memberof HomeScreen
   * @description TDB
   */
  async function _createFirstStampCard() {
    try {
      var newStampCard: StampCardType = {
        title: '',
        complete: false,
        date_of_creation: new Date(),
      };

      var card: StampCardType = await StampCards.insert(newStampCard, true)[0];
      var newStamps: StampType[] = [];

      for (let i = 1; i <= 10; i++) {
        var stamp: StampType = {
          name: undefined,
          image: undefined,
          description: undefined,
          date: undefined,
          picture: undefined,
          number: i,
          done: false,
          stampCard: card,
        };
        newStamps.push(stamp);
      }

      await Stamps.insert(newStamps, true);
    } catch (error) {
      throw new Error('_createFirstStampCard' + ' ' + error.message);
    }
  }

  function RenderAccordionItems() {
    return (
      <Accordion
        key={Math.random()}
        bg={mainBgColor}
        safeAreaX
        index={[0]}
        defaultIndex={[defaultIndex]}
        borderColor="transparent">
        {cards.map((card) => (
          <Accordion.Item key={card.id}>
            <Accordion.Summary
              bg={mainBgColor}
              _text={{
                color: accordionTextColorMode,
              }}
              _expanded={{
                backgroundColor: mainBgColor,
                _text: {
                  color: accordionTextColorMode,
                },
              }}>
              {card.title !== '' ? card.title : 'Aktuelle Stempelkarte'}
              <Accordion.Icon color={accordionTextColorMode} />
            </Accordion.Summary>
            <Accordion.Details>
              <StampCard item={card} />
            </Accordion.Details>
          </Accordion.Item>
        ))}
      </Accordion>
    );
  }

  return (
    <View bg={mainBgColor} justifyContent="center" alignContent="center">
      <TourGuideZone
        zone={1}
        shape="circle"
        text={strings('TOURGUIDE_HEADER_TEXT')}
        borderRadius={25}>
        <HomeScreenHeader />
      </TourGuideZone>

      {/* <Button
        alignSelf="flex-start"
        size="md"
        variant="ghost"
        onPress={() => {
          toggleColorMode();
        }}>
        Change mode
      </Button> */}
      <ScrollView
        // eslint-disable-next-line react-native/no-inline-styles
        _contentContainerStyle={{
          justifyContent: 'center',
          bg: mainBgColor,
        }} // style={{ backgroundColor: 'blue' }}
      >
        <Stack direction={'column'} space={5}>
          <TourGuideZone
            zone={2}
            shape={'circle'}
            text={strings('TOURGUIDE_MEMBERCARD_TEXT')}
            borderRadius={25}>
            <MemberCard />
          </TourGuideZone>

          <TourGuideZone
            zone={3}
            shape={'circle'}
            text={strings('TOURGUIDE_STAMPCARD_TEXT')}
            borderRadius={25}>
            <RenderAccordionItems />
          </TourGuideZone>
        </Stack>
        <Box safeArea h={10} />
      </ScrollView>
    </View>
  );
};

export default HomeScreen;
