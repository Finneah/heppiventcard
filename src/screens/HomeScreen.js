import React, {useRef, useState} from 'react';
import {Accordion, Card, CardItem, Container, Content} from 'native-base';
import {Dimensions, Linking, StyleSheet} from 'react-native';
import {
  TourGuideZone, // Main wrapper of highlight component
  useTourGuideController, // hook to start, etc.
} from 'rn-tourguide';
import GlobalColors from '../styles/GlobalColors';
import HomeScreenHeader from '../components/HomeScreenHeader';
import MemberCard from '../components/MemberCard';
import StampCard from '../components/StampCard';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {StampCards, Stamps} from '../database';

import {StampsModel} from '../database/Models/StampsModel';
import {StampCardsSchema} from '../database/Schemas/StampCardsSchema';
import {strings} from '../i18n';
let stampCardsSchema = new StampCardsSchema();
let stampsModel = new StampsModel();

const numCol = 3;
const TEST = false;

/**
 * @category View
 * @namespace HomeScreen
 * @description Startpage of the app.
 */
const HomeScreen = ({navigation}) => {
  let _accordion = useRef(null);

  const {
    canStart, // a boolean indicate if you can start tour guide
    start, // a function to start the tourguide
    stop, // a function  to stopping it
    eventEmitter, // an object for listening some events
  } = useTourGuideController();
  const [cards, setCards] = useState([]);

  const handleOnStop = () => {
    _setTutorialIsDone();
  };

  React.useLayoutEffect(() => {
    StampCards.onLoaded(async () => {
      if (TEST) {
        console.info('StampCards loaded', StampCards.data());
      }
      if (StampCards.data().length === 0) {
        await _createFirstStampCard();
      }
    });

    StampCards.onChange(async () => {
      if (TEST) {
        console.info('StampCards changed', StampCards.data());
      }
    });

    Stamps.onLoaded(async () => {
      if (TEST) {
        console.info('Stamps loaded', Stamps.data());
      }
      _getStampCards(StampCards.data());
    });

    Stamps.onChange(async () => {
      if (TEST) {
        console.info('Stamps loaded', Stamps.data());
      }
      _getStampCards(StampCards.data());
    });

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  React.useEffect(() => {
    _startIfPossible();

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [canStart]); // ðŸ‘ˆ don't miss it!

  React.useEffect(() => {
    eventEmitter.on('stop', handleOnStop);

    return () => {
      eventEmitter.off('stop', handleOnStop);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  async function _startIfPossible() {
    var tutorialDone = await _getTutorialIsDone();

    if (canStart && !tutorialDone) {
      // ðŸ‘ˆ test if you can start otherwise nothing will happen
      start();
    }
  }
  async function _setTutorialIsDone() {
    try {
      await AsyncStorage.setItem('@tutorialDone', 'true');
    } catch (e) {
      console.warn(e);
    }
  }

  async function _getTutorialIsDone() {
    try {
      const value = await AsyncStorage.getItem('@tutorialDone');
      if (value !== null) {
        return true;
      } else {
        return false;
      }
    } catch (e) {
      console.warn(e);
    }
  }

  async function _getAccordionExpandedId() {
    try {
      const value = await AsyncStorage.getItem('@accordionExpandedId');
      if (value !== null) {
        return value;
      }
    } catch (e) {
      console.warn(e);
    }
  }
  /**
   * @memberof HomeScreen
   * @param {Object} sectionStampCards
   */
  async function _getStampCards(sectionStampCards) {
    var sections = [];
    var expandedId = await _getAccordionExpandedId();

    if (TEST) {
      console.info('_getStampCards', sectionStampCards, Stamps.data());
    }
    sectionStampCards.forEach(async (stampCard) => {
      var sectionPart = _setSectionPart(stampCard);
      sections.push(sectionPart);
    });
    sections = sections.reverse();
    setCards(sections);

    _setSelectedAccordionItem(sections, expandedId);
  }

  function _setSectionPart(stampCard) {
    var sectionPart = stampCard;
    sectionPart.title =
      stampCard.title === '' ? 'Aktuelle Stempelkarte' : stampCard.title;
    var content = stampsModel.filterStampsBy({
      stampCard_id: stampCard.id,
    });
    if (content.length === 0) {
      content = stampsModel.filterStampsBy({
        stampCard: stampCard.id,
      });
    }

    sectionPart.content = content;
    return sectionPart;
  }

  function _setSelectedAccordionItem(sections, expandedId) {
    for (let i = 0; i < sections.length; i++) {
      const element = sections[i];
      if (expandedId === element.id) {
        _accordion.current.setSelected(i);
      }
    }
  }
  /**
   * @memberof HomeScreen
   */
  async function _createFirstStampCard() {
    var props = stampCardsSchema.props;

    var newStampCard = {};
    for (const key in props) {
      if (Object.hasOwnProperty.call(props, key)) {
        newStampCard[key] = undefined;
      }
    }

    newStampCard.date_of_creation = new Date();
    newStampCard.title = '';
    newStampCard.complete = false;

    var card = await StampCards.insert(newStampCard, true)[0];

    var newStamps = [];
    for (let i = 1; i <= 10; i++) {
      newStamps.push({
        number: i,
        done: 0,
        stampCard: card,
      });
    }

    await Stamps.insert(newStamps, true);
  }

  return (
    <Container style={{padding: 5}}>
      <TourGuideZone
        zone={1}
        shape="circle"
        text={strings('TOURGUIDE_HEADER_TEXT')}
        borderRadius={25}>
        <HomeScreenHeader
          onPress={() => {
            Linking.openURL('http://heppivents.de/').catch((err) =>
              console.error('An error occured', err),
            );
          }}
        />
      </TourGuideZone>

      <Content>
        <TourGuideZone
          zone={2}
          shape={'circle'}
          text={strings('TOURGUIDE_MEMBERCARD_TEXT')}
          borderRadius={25}>
          <MemberCard />
        </TourGuideZone>
        <Card>
          <CardItem first last>
            <TourGuideZone
              zone={3}
              shape={'circle'}
              text={strings('TOURGUIDE_STAMPCARD_TEXT')}
              borderRadius={25}>
              <Accordion
                ref={_accordion}
                style={styles.transparentBorder}
                dataArray={cards}
                animation={true}
                expanded={0}
                onAccordionOpen={async (item, index) => {
                  var expandedId = await _getAccordionExpandedId();
                  if (item.id !== expandedId) {
                    await AsyncStorage.removeItem('@accordionExpandedId');
                  }
                }}
                headerStyle={[styles.transparentBG]}
                icon="chevron-forward"
                expandedIcon="chevron-down"
                iconStyle={{color: GlobalColors.brandPrimary}}
                expandedIconStyle={{color: GlobalColors.brandSecondary}}
                renderContent={(item) => {
                  if (TEST) {
                    console.info('renderContent', item);
                  }

                  return <StampCard item={item} />;
                }}
              />
            </TourGuideZone>
          </CardItem>
        </Card>
      </Content>
    </Container>
  );
};

const styles = StyleSheet.create({
  item: {
    flexDirection: 'column',
    backgroundColor: 'transparent',
    margin: 5,
    padding: 0,
    borderRadius: 20,

    width: Dimensions.get('screen').width * ((100 / numCol - 10) / 100),
    height: Dimensions.get('screen').width * ((100 / numCol - 10) / 100),
  },
  content: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignContent: 'space-between',
    justifyContent: 'space-evenly',
  },
  lastItem: {
    backgroundColor: GlobalColors.lightGrey,
    alignItems: 'center',
    justifyContent: 'center',
    width: Dimensions.get('screen').width * ((100 / numCol - 10) / 100) * 2.2,
  },
  image: {
    backgroundColor: GlobalColors.lightGrey,
    borderRadius: 20,
    width: '100%',
    height: '100%',
    justifyContent: 'center',
  },
  stampItem: {
    borderRadius: 20,
    justifyContent: 'center',
    alignItems: 'center',
  },
  transparentBorder: {borderColor: 'transparent'},
  transparentBG: {backgroundColor: 'transparent'},
});

export default HomeScreen;
