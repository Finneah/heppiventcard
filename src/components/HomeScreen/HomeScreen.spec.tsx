import 'react-native';
import * as React from 'react';
import {render} from '@testing-library/react-native';

// Note: test renderer must be required after react-native.

import {NativeBaseProvider} from 'native-base';
import HomeScreen from './HomeScreen';

const wrapper = ({children}: any) => (
  <NativeBaseProvider
    initialWindowMetrics={{
      frame: {x: 0, y: 0, width: 0, height: 0},
      insets: {top: 0, left: 0, right: 0, bottom: 0},
    }}>
    {children}
  </NativeBaseProvider>
);

describe('HomeScreen', () => {
  test('renders correctly', () => {
    const homeScreen = render(<HomeScreen />, {wrapper});

    expect(homeScreen.toJSON()).toMatchSnapshot();
  });
});
