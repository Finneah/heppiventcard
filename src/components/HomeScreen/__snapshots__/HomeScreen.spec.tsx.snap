// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    _text={
      Object {
        "color": "darkText",
      }
    }
    alignContent="center"
    bg="white"
    justifyContent="center"
    style={
      Array [
        Object {
          "alignContent": "center",
          "backgroundColor": "#FFFFFF",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
    >
      <View
        alignItems="flex-end"
        bg="white"
        flexDirection="row"
        justifyContent="space-between"
        marginBottom={10}
        px={2}
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "backgroundColor": "#FFFFFF",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 40,
              "paddingLeft": 8,
              "paddingRight": 8,
            },
          ]
        }
      >
        <Text
          color="muted.700"
          fontFamily="body"
          fontSize="xl"
          fontStyle="normal"
          fontWeight="bold"
          style={
            Array [
              Object {
                "color": "#404040",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "700",
              },
            ]
          }
        >
          APPTITLE
        </Text>
        <View
          style={
            Array [
              Object {},
            ]
          }
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onHoverIn={[Function]}
          onHoverOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {},
            ]
          }
          testID="pressableLogo"
        >
          <Image
            accessibilityLabel="heppiventlogo, click to open the website"
            alt="heppiventlogo, click to open the website"
            maxWidth="100%"
            onError={[Function]}
            source={
              Object {
                "testUri": "../../../src/image/logo.jpg",
              }
            }
            style={
              Array [
                Object {
                  "maxWidth": "100%",
                },
                Object {
                  "height": 50,
                  "justifyContent": "center",
                  "resizeMode": "stretch",
                  "width": 120,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "backgroundColor": "#FFFFFF",
          "justifyContent": "center",
        }
      }
      style={
        Array [
          Object {},
        ]
      }
    >
      <View>
        <View
          flexDirection="column"
          style={
            Array [
              Object {
                "flexDirection": "column",
              },
            ]
          }
        >
          <View
            onLayout={[Function]}
          >
            <View
              alignItems="center"
              flexDirection="row"
              justifyContent="space-between"
              px={5}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingLeft": 20,
                    "paddingRight": 20,
                  },
                ]
              }
            >
              <View
                alignItems="center"
                flexDirection="row"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  fontSize="24px"
                  lineHeight="24px"
                  onPress={[Function]}
                  style={
                    Array [
                      Object {
                        "color": "primary.600",
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#0891b2",
                          "fontSize": 24,
                          "height": 24,
                          "lineHeight": 24,
                          "width": 24,
                        },
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ï‰«
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "width": 8,
                      },
                    ]
                  }
                  width={2}
                />
                <Text
                  color="muted.700"
                  fontFamily="body"
                  fontSize="md"
                  fontStyle="normal"
                  fontWeight="bold"
                  style={
                    Array [
                      Object {
                        "color": "#404040",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                      },
                    ]
                  }
                >
                  HELLO NAME_PLACEHOLDER
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              />
              <Text
                color="secondary.400"
                fontFamily="body"
                fontSize="md"
                fontStyle="normal"
                fontWeight="bold"
                style={
                  Array [
                    Object {
                      "color": "#f472b6",
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                    },
                  ]
                }
              >
                YOUR_RANK RANK_00
              </Text>
            </View>
          </View>
          <View
            height={5}
            style={
              Array [
                Object {
                  "height": 20,
                },
              ]
            }
          />
          <View
            onLayout={[Function]}
          >
            <View
              bg="white"
              borderColor="transparent"
              borderRadius="lg"
              borderWidth={1}
              overflow="hidden"
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "transparent",
                    "borderRadius": 6,
                    "borderWidth": 1,
                    "overflow": "hidden",
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          h={10}
          style={
            Array [
              Object {
                "height": 40,
              },
            ]
          }
        />
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;
